app/uk/gov/hmrc/tradergoodsprofilesdatastore/controllers/GetRecordsController.scala

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.tradergoodsprofilesdatastore.controllers
+
+import play.api.libs.Files.logger
+import play.api.libs.json.Json
+import play.api.mvc.{Action, AnyContent, ControllerComponents}
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.play.bootstrap.backend.controller.BackendController
+import uk.gov.hmrc.tradergoodsprofilesdatastore.connectors.RouterConnector
+import uk.gov.hmrc.tradergoodsprofilesdatastore.controllers.actions.IdentifierAction
+import uk.gov.hmrc.tradergoodsprofilesdatastore.models.GetRecordsResponse
+import uk.gov.hmrc.tradergoodsprofilesdatastore.repositories.RecordsRepository
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class GetRecordsController @Inject() (
+  recordsRepository: RecordsRepository,
+  cc: ControllerComponents,
+  routerConnector: RouterConnector,
+  identify: IdentifierAction
+)(implicit ec: ExecutionContext)
+    extends BackendController(cc) {
+
+  def getRecords(
+    eori: String,
+    lastUpdatedDate: Option[String],
+    page: Option[Int],
+    size: Option[Int]
+  ): Action[AnyContent] = identify.async { implicit request =>
+    implicit val hc: HeaderCarrier = HeaderCarrier().withExtraHeaders("X-Client-ID" -> "TGP-Frontend")
+
+    routerConnector
+      .getRecords(eori, lastUpdatedDate, page, size)(hc)
+      .flatMap {
+        case httpResponse if httpResponse.status == OK =>
```

Comment:
I think it will throw an exception if it's not a success state so probably don't need to worry about matching here. If an exception is thrown, let it bubble out to the default error handler rather than catching and returning 500 here because we want to return an error page rather than a body with an error message